app/service.js

```diff
@@ -58,11 +50,19 @@ app.post('/v1/metadata', function(req, res) {
   const promises = req.body.urls.map((url) => getUrlMetadata(url));
 
   Promise.all(promises)
-    .then((urlsData) => {
-      responseData.urls = buildObj(urlsData.map((urlData) => [urlData.url, urlData]));
+    .then((results) => {
+      results.map((result) => {
```

Comment:
Super ultra nit: You could also probably destructure the `{error, url, data}` here and save a few keystrokes below.